code-studio/src/js/initApp/initApp.js

```diff
@@ -169,8 +169,8 @@ window.apps = {
         // If we're readOnly, source hasn't changed at all
         source = Blockly.mainBlockSpace.isReadOnly() ? currentLevelSource :
           Blockly.Xml.domToText(Blockly.Xml.blockSpaceToDom(Blockly.mainBlockSpace));
-      } else {
-        source = window.Applab && Applab.getCode();
+      } else if (appOptions.getCode) {
+        source = appOptions.getCode();
```

Comment:
Took me a little while to figure it out. I still have a shared directory which I think I should be ignoring. And I edited generated code a few times before I realized how it all worked!